github/client.go

```diff
@@ -970,6 +985,29 @@ func (client *Client) apiClient() *simpleClient {
 	return &simpleClient{
 		httpClient: httpClient,
 		rootUrl:    apiRoot,
+		extraHeader: func(r *http.Header) {
+			if headers, err := git.ConfigAll(fmt.Sprintf("http.%s://%s.extraheader", client.Host.Protocol, client.Host.Host)); err == nil {
+				url, _ := url.Parse(fmt.Sprintf("%s://%s", client.Host.Protocol, client.Host.Host))
+				cookies := []*http.Cookie{}
+
+				for _, header := range headers {
+					keyValue := strings.Split(header, ":")
+					key, value := strings.TrimSpace(keyValue[0]), strings.TrimSpace(keyValue[1])
+
+					if strings.ToUpper(key) == "COOKIE" && strings.Contains(value, "=") {
+						cookieKV := strings.Split(value, "=")
+						cookieKey, cookieValue := strings.TrimSpace(cookieKV[0]), strings.TrimSpace(cookieKV[1])
+						cookies = append(cookies, &http.Cookie{Name: cookieKey, Value: cookieValue})
+						// r.Add(key,value)
+					}
+				}
+
+				if len(cookies) > 0 && httpClient.Jar == nil {
+					httpClient.Jar, _ = cookiejar.New(&cookiejar.Options{PublicSuffixList: publicsuffix.List})
+					httpClient.Jar.SetCookies(url, cookies)
```

Comment:
Is keeping a cookie jar really necessary for API requests to your Enterprise instance?

Reply:
The dataflow is:
1. Nginx
1. 302: redirect to KeyCloack
   KeyCloack is forwarding to internal server with SessionID and traceIds
1. Client is requesting again the GHE behind Nginx 
1. Nginx is passing to GHE

So there are 2 or 3 round trips with a bunch of cookies